develop branch

TimurSharibov opened the pull request.

TimurSharibov committed:
develop branch

TimurSharibov committed:
Utf-8

pomegranate282922 commented on src/test/java/helpers/CourierHelper.java:
```diff
@@ -1,2 +1,28 @@
+        return given()
+                .contentType(ContentType.JSON)
+                .body(courier)
+                .post("/api/v1/courier");
```
Урлы выносим как константы 
Нужно исправить

pomegranate282922 commented on src/test/java/helpers/CourierHelper.java:
```diff
@@ -1,2 +1,28 @@
+                .post("/api/v1/courier/login");
+    }
+
+    public static Response deleteCourier(int id) {
```
Нужно расставить степы 
Нужно исправить

pomegranate282922 commented on src/test/java/tests/CourierTests.java:
```diff
@@ -1,2 +1,69 @@
+
+    @Before
+    public void setUp() {
+        courier = new Courier("ninja", "1234", "saske");
```
Лучше использовать рандомизацию 
Можно улучшить

pomegranate282922 commented on src/test/java/tests/CourierTests.java:
```diff
@@ -1,2 +1,69 @@
+        courier = new Courier("ninja", "1234", "saske");
+    }
+
+    @Step("�������� �������")
```
Это не степ. Нужно дать аллюр описание тесту 
Нужно исправить

pomegranate282922 commented on src/test/java/tests/CourierTests.java:
```diff
@@ -1,2 +1,69 @@
+
+    @After
+    public void tearDown() {
+        if (courierId != 0) {
```
Не везде удаляется созданный курьер 
Нужно исправить

pomegranate282922 commented on src/test/java/tests/OrderTests.java:
```diff
@@ -1,2 +1,36 @@
+
+    @Before
+    public void setUp() {
+        order = new Order("Naruto", "Uchiha", "Konoha, 142 apt.", 4, "+7 800 355 35 35", 5, "2020-06-06", "Saske, come back to Konoha", Arrays.asList("BLACK"));
```
Нужно проверить все 4 варианта цветов - используя параметризацию 
Нужно исправить

pomegranate282922 commented on src/test/java/tests/OrderTests.java:
```diff
@@ -1,2 +1,36 @@
+    @Test
+    public void getOrderList() {
+        Response response = OrderHelper.getOrderList();
+        response.then().statusCode(200).and().body("orders", notNullValue());
```
Лучше проверить непустоту списка 
Можно улучшить

pomegranate282922 commented on src/test/java/utils/RestAssuredConfig.java:
```diff
@@ -1,2 +1,11 @@
+public class RestAssuredConfig {
+    @BeforeClass
+    public static void setup() {
+        RestAssured.baseURI = Config.BASE_URL;
```
Отлично что есть иерархия тестов

pomegranate282922 reviewed (commented):
Работа хорошая! 
замечания:
1. Использовать параметризацию в создании заказа
2. Вынести все урлы как константы 
3. Расставить степы с именами 
4. Всем тестам дать аллюр описания 
5. Везде проверить чтобы удалялся созданный курьер 
6. Нет аллюр отчета 
7. Остальное по коду

TimurSharibov committed:
Исправлены замечания:
Использовать параметризацию в создании заказа
Вынести все урлы как константы
Расставить степы с именами
Всем тестам дать аллюр описания
Везде проверить чтобы удалялся созданный курьер
Остальное по коду

TimurSharibov committed:
Исправлен pom.xml для аллюр отчета. Тесты все рабочие.

TimurSharibov committed:
Добавлен аллюр отчет.

pomegranate282922 commented on src/test/java/tests/CourierTests.java:
```diff
@@ -1,2 +1,75 @@
+    @Test
+    @Description("�������� ������ ��� �������� ������������ �������")
+    public void createDuplicateCourier() {
+        CourierHelper.createCourier(courier);
```
Не удаляется курьер 
Нужно исправить

pomegranate282922 reviewed (commented):
Многое исправил- супер! Осталось:
1. Не везде удаляется курьер 
2. Нет аллюр отчета

TimurSharibov committed:
Добавлен новый аллюр отчет.

TimurSharibov committed:
Добавлен новый аллюр отчет. Теперь удаляется курьер во всех тестах и проверяется на пустое id курьера.

TimurSharibov committed:
Добавлен новый аллюр отчет. Теперь удаляется курьер во всех тестах и проверяется на пустое id курьера.

pomegranate282922 reviewed (approved):
Ок, зачет!
